web/src/loading.ts

```diff
@@ -92,3 +92,27 @@ export function show_button_spinner($elt: JQuery, using_dark_theme: boolean): vo
     }
     $elt.css("display", "inline-block");
 }
+
+// For showing spinners at dialog_widget and user_profile.
+export function show_spinner($button: JQuery): void {
+    const $spinner = $button.find(".modal__spinner");
+    const dialog_submit_button_span_width = $button.find("span").width();
+    const dialog_submit_button_span_height = $button.find("span").height();
+
+    // Disable buttons.
+    $button.prop("disabled", true);
+    $button.find("span").hide();
+
+    // Show spinner.
+    make_indicator($spinner, {
+        width: dialog_submit_button_span_width,
+        height: dialog_submit_button_span_height,
+    });
+}
```

Comment:
While we want to disable both the modal buttons (`.dialog_submit_button` and `.dialog_exit_button`). We only want to hide the the `span` of the `.dialog_submit_button`. But in almost all of the times you have called `show_spinner()`, you have passed the query selector as `#dialog_widget_modal .modal__btn`, which selects both the submit and the cancel buttons -- something we should definitely avoid, given we don't want `show_spinner` being called on the cancel button as well (saved currently by the lack of any `span` or `.modal__spinner` elements).